packages/connect/src/connector/smoldot-light.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import type {
+  Chain as SChain,
+  Client,
+  ClientOptions,
+  AlreadyDestroyedError as IAlreadyDestroyedError,
+  CrashError as ICrashError,
+  JsonRpcDisabledError as IJsonRpcDisabledError,
+} from "@substrate/smoldot-light"
+import {
+  AlreadyDestroyedError,
+  CrashError,
+  JsonRpcDisabledError,
+} from "./errors.js"
+import { getSpec } from "./specs/index.js"
+import type {
+  AddChain,
+  AddWellKnownChain,
+  Chain,
+  SubstrateConnector,
+} from "./types.js"
+import { SupportedChains } from "../SupportedChains.js"
+
+let SdAlreadyDestroyedError: typeof IAlreadyDestroyedError
+let SdCrashError: typeof ICrashError
+let SdJsonRpcDisabledError: typeof IJsonRpcDisabledError
+
+let startPromise: Promise<(options: ClientOptions) => Client> | null = null
+const getStart = () => {
+  if (startPromise) return startPromise
+  return (startPromise = import("@substrate/smoldot-light").then((sm) => {
+    SdJsonRpcDisabledError = sm.JsonRpcDisabledError
+    SdCrashError = sm.CrashError
+    SdAlreadyDestroyedError = sm.AlreadyDestroyedError
+    return sm.start
+  }))
+}
+
+let totalActiveChains = 0
+let clientPromise: Promise<Client> | null = null
+const getClient = (): Promise<Client> => {
+  if (clientPromise) return clientPromise
+  return (clientPromise = getStart().then((start) =>
+    start({
+      forbidNonLocalWs: true, // Prevents browsers from emitting warnings if smoldot tried to establish non-secure WebSocket connections
+      maxLogLevel: 3 /* no debug/trace messages */,
+    }),
+  ))
```

Comment:
```suggestion
  clientPromise = getStart().then((start) =>
    start({
      forbidNonLocalWs: true, // Prevents browsers from emitting warnings if smoldot tried to establish non-secure WebSocket connections
      maxLogLevel: 3 /* no debug/trace messages */,
    }),
  )
  return clientPromise
```

We want to make code as readable as possible